src/lib.rs

```diff
@@ -248,31 +258,307 @@ impl<T, const N: usize> ArrayVec<T, { N }> {
+        // Write it in, overwriting the first copy of the `index`th
+        // element.
+        ptr::write(p, item);
+    }
+
+    /// Remove the value contained at `index` and return it.
+    ///
+    /// # Panics
+    ///
+    /// The index is out of bounds.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use const_arrayvec::ArrayVec;
+    /// let mut vector = ArrayVec::from([4, 3, 2]);
+    ///
+    /// let three = vector.remove(1);
+    ///
+    /// assert_eq!(three, 3);
+    /// assert_eq!(&vector, [4, 2].as_ref());
+    /// ```
+    pub fn remove(&mut self, index: usize) -> T {
+        if let Some(item) = self.try_remove(index) {
+            item
+        } else {
+            out_of_bounds!("remove", index, self.len());
+        }
+    }
+
+    /// If `index` is in bounds, remove the value contained at `index`
+    /// and return it.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use const_arrayvec::ArrayVec;
+    /// let mut vector = ArrayVec::from([4, 3, 2]);
+    ///
+    /// let three = vector.try_remove(1);
+    /// let what = vector.try_remove(24);
+    ///
+    /// assert_eq!(three, Some(3));
+    /// assert_eq!(what, None);
+    /// assert_eq!(&vector, [4, 2].as_ref());
+    /// ```
+    #[inline]
+    pub fn try_remove(&mut self, index: usize) -> Option<T> {
+        if index < self.len() {
+            Some(unsafe { self.remove_unchecked(index) })
+        } else {
+            None
+        }
+    }
+
+    /// Remove the value contained at `index` and return it.
+    ///
+    /// # Safety
+    ///
+    /// The index must be in bounds.
+    pub unsafe fn remove_unchecked(&mut self, index: usize) -> T {
+        let len = self.len();
+
+        // Where the value to remove is.
+        let p = self.as_mut_ptr().add(index);
+        // Read the value before sending it to the other world.
+        let item = ptr::read(p);
+        // Shift every value after the removed one to the left.
+        ptr::copy(p.add(1), p, len - index - 1);
+        // We removed an item, so the length should be decremented.
+        self.set_len(len - 1);
+
+        item
+    }
+
+    /// Remove the value contained at `index` and return it without
+    /// conserving order.
+    ///
+    /// The removed value is replaced by the last value making this an
+    /// `O(1)` operation.
+    ///
+    /// # Panics
+    ///
+    /// The index is out of bounds.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use const_arrayvec::ArrayVec;
+    /// let mut vector = ArrayVec::from([1, 2, 4]);
+    ///
+    /// assert_eq!(vector.swap_remove(0), 1);
+    /// assert_eq!(&vector, [4, 2].as_ref());
+    ///
+    /// assert_eq!(vector.swap_remove(1), 2);
+    /// assert_eq!(&vector, [4].as_ref());
+    ///
+    /// assert_eq!(vector.swap_remove(0), 4);
+    /// assert_eq!(vector.len(), 0);
+    /// ```
+    pub fn swap_remove(&mut self, index: usize) -> T {
+        if let Some(item) = self.try_swap_remove(index) {
```

Comment:
Could this be better written as a `match` statement? Also, what made you want to go through `try_swap_remove()` instead of doing the bounds check and calling `swap_remove_unchecked()` directly?

Reply:
Sure, I'll change it into a `match`.
As for using `try_swap_remove` instead of `swap_remove_unchecked` that's because there is already a function that performs all the bound checks while also being safe so it felt right to use it.
However, I don't have any problems with rewriting this to use `remove_unchecked` directly instead, the final decision's yours!